src/Application/Teacher/Commands/CreateTeacher/CreateTeacherCommandHandler.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Application.Abstractions;
+using Application.Abstractions.IPersistence;
+using Application.Common.Results;
+using Application.Teacher.Dtos;
+using Application.Teacher.Repository;
+using Domain.Abstractions.Enum;
+using Domain.Entities;
+using FluentValidation;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Teacher.Commands.CreateTeacher
+{
+    public sealed class CreateTeacherCommandHandler 
+        : IRequestHandler<CreateTeacherCommand, OperationResult<CreateTeacherDto>>
+    {
+        private readonly ITeacherRepository _teacherRepository;
+        private readonly IPasswordHasher _passwordHasher;
+        private readonly IValidator<CreateTeacherCommand> _validator;
+        private readonly IAppDbContext _db;
+        
+        public CreateTeacherCommandHandler(
+            ITeacherRepository teacherRepository,
+            IPasswordHasher passwordHasher,
+            IValidator<CreateTeacherCommand> validator,
+            IAppDbContext db)
+        {
+            _teacherRepository = teacherRepository;
+            _passwordHasher = passwordHasher;
+            _validator = validator;
+            _db = db;
+        }
+
+        public async Task<OperationResult<CreateTeacherDto>> Handle(CreateTeacherCommand request,
+            CancellationToken ct)
+        {
+            var validationResult = await _validator.ValidateAsync(request);
+            if (!validationResult.IsValid)
+            {
+                var error = Error.Validation("Validation.Error", string.Join(",", validationResult.Errors.Select(e => e.ErrorMessage)));
+                return OperationResult<CreateTeacherDto>.Failure(error);
+            }
+            
+            var email = request.Email.Trim().ToLowerInvariant();
+            if (await _db.Users.AnyAsync(u => u.Email == email, ct))
```

Comment:
Du bryter ditt repository pattern i och med att du gör anrop direkt till databasen härifrån för att kolla om användare finns men när du lägger till användare gör du detta genom ditt repository. Porque?

Jag tycker att denna logic bör vara i ditt repository lager. Alla databas anrop du gör var sig läs/skriv ska göras ifrån ditt repository. 1 single point of access.